File: KT10/src/main/java/net/println/kt10/LazyNotThreadSafe.java
package net.println.kt10;

/**
 * Created by luliju on 2017/7/8.
 */
public class LazyNotThreadSafe {
    private static LazyNotThreadSafe INSTANCE;

    private LazyNotThreadSafe(){}

    public static LazyNotThreadSafe getInstance(){
        if(INSTANCE == null){
            INSTANCE = new LazyNotThreadSafe();
        }
        return INSTANCE;
    }
}

File: KT10/src/main/kotlin/net/println/kt10/LazyNotThreadSafe.kt
package net.println.kt10.kotlin

/**
 * Created by luliju on 2017/7/8.
 */
class LazyNotThreadSafe {

    companion object{
        val instance by lazy(LazyThreadSafetyMode.NONE) {
            LazyNotThreadSafe()
        }

        //下面是另一种等价的写法, 获取单例使用 get 方法
        private var instance2: LazyNotThreadSafe? = null

        fun get() : LazyNotThreadSafe {
            if(instance2 == null){
                instance2 = LazyNotThreadSafe()
            }
            return instance2!!
        }
    }
}